Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -197,9 +235,10 @@ extension MapboxMap: ObservableProtocol {
 // MARK: - Map Event handling
 
 extension MapboxMap: MapEventsObservable {
-    @discardableResult
-    public func onNext(_ eventType: MapEvents.EventKind, handler: @escaping (Event) -> Void) -> Cancelable {
-        let handler = MapEventHandler(for: [eventType.rawValue],
+
+    private func onNext(eventTypes: [MapEvents.EventKind], handler: @escaping (Event) -> Void) -> Cancelable {
```

Comment:
Why did we remove the `@discardableResult` The use case above doesn't use the result so can we still go with discarding it?

Reply:
This is the private function, and wanted to be a little more explicit in this case. If we make this other function public let's add then?

Reply:
I feel like because its private, that gives us more reasons to add @discardableResult. Or maybe even remove the return type

Reply:
Done, and added tests.